src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -347,7 +349,7 @@ async def search_dataset_records(
 
     for record in records:
         record_id_score_map[record.id]["search_record"] = SearchRecord(
-            record=record, query_score=record_id_score_map[record.id]["query_score"]
+            record=record.__dict__, query_score=record_id_score_map[record.id]["query_score"]
```

Comment:
I think this is needed regarding the empty `responses` error, right?

Reply:
It won't give an error, but if we pass `record` instead of `record.__dict__`, `pydantic` tries to get the value of `record.responses`, and in this case `sqlalchemy` will return an empty list.

`record.__dict__` doesn't contain the `responses` key, and this way the value in the `pydantic.BaseModel` will be `None` (because it's an `Optional` field), which in combination with the `response_model_exclude_unset` endpoint flag makes the endpoint to not include the `responses` key in the output payload, which is the desired behaviour (instead of receiving `"responses": []`.